src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -411,12 +411,25 @@ open_locale arithmetic_function
 
 /-- `cyclotomic n R` can be expressed as a product in a fraction field of `polynomial R`
   using Möbius inversion. -/
-lemma cyclotomic_eq_prod_X_pow_sub_one_pow_moebius {n : ℕ} (hpos : 0 < n)
-  (R : Type*) [comm_ring R] [nontrivial R]
+lemma cyclotomic_eq_prod_X_pow_sub_one_pow_moebius {n : ℕ} (R : Type*) [comm_ring R]
   {K : Type*} [field K] [algebra (polynomial R) K] [is_fraction_ring (polynomial R) K] :
   algebra_map _ K (cyclotomic n R) =
     ∏ i in n.divisors_antidiagonal, (algebra_map (polynomial R) K (X ^ i.snd - 1)) ^ μ i.fst :=
```

Comment:
I think we should just close this PR and open a separate one; also, @riccardobrasca what do you think of this change?

Reply:
LGTM too. Can we evaluate a rational function? If we already have that, the (currently not so nice) proof of `polynomial.cyclotomic_coeff_zero` should be golfed.

Reply:
Yes, [`ratfunc.eval`](https://github.com/leanprover-community/mathlib/blob/master/src/field_theory/ratfunc.lean#L742) should work as expected (except when dividing by 0 of course!)

Reply:
we can't golf the proof, sadly, as `coeff_zero` works for all `comm_rings`

Reply:
I think we can prove it for `Z` and then use that `cyclotomic n R` comes from `cyclotomic n Z`.

Reply:
I keep missing this and just saying "can't be done"! need to turn on my brain instead of just being in Lean mode all the time. Anyways, I currently have a sorry-free proof of `∀x, eval (cyclotomic n ℝ) x > 0` waiting for #10422, which is actually necessary for this proof (`ratfunc.eval` is only multiplicative if there isn't division by zero). There's also some issues with `ratfunc ℤ.eval`, but these can be easily dealt with by going to ℚ instead so there's no problems there.

Reply:
Well, mathematically there is only `cyclotomic n Z`. In Lean it is convenient to have `cyclotomic n R`, but it comes from `Z` by definition. If something is wrong with integers coefficients we should fix it

Reply:
So if the constant term of `cyclotomic n Z` is `1` the same is true for `cyclotomic n `

Reply:
yes I agree, I'm just laying out some of the issues that happen in Lean; there's nothing wrong with integer coeffs, just it's far more convenient to work over a field with `ratfunc`, and the suffices work out so that we do indeed get the result for all `R`.